lib/run/summary.js

```diff
@@ -223,14 +233,22 @@ _.assign(RunSummary, {
         emitter.on('request', function (err, o) {
             if (!_.get(o, 'cursor.ref')) { return; }
 
-            var execution = cache[o.cursor.ref] = (cache[o.cursor.ref] || {});
-
-            executions.push(_.assignIn(execution, {
-                cursor: o.cursor,
-                request: o.request,
-                response: o.response,
-                id: _.get(o, 'item.id')
-            }, err && {
+            var execution = cache[o.cursor.ref] = (cache[o.cursor.ref] || {}),
+                executionResponse = {
+                    cursor: o.cursor,
+                    request: o.request,
+                    response: o.response,
+                    id: _.get(o, 'item.id')
+                };
+
+            // check if skipExecutionResponse set in options or if cli is the only reporter specified, we do
+            // not include executionResponse as part of executions summary
+            if (summary.skipExecutionResponse || (_.isArray(summary.reporters) &&
+                summary.reporters.length === 1 && (summary.reporters).includes('cli'))) {
```

Comment:
We can put the value of whether the only reporter is `cli` inside a variable for readability:
```suggestion
			var hasOnlyCliReporter = _.isArray(summary.reporters) &&
		    	summary.reporters.length === 1 &&
		    	summary.reporters.includes('cli');
    	
            if (summary.skipExecutionResponse || hasOnlyCliReporter) {
```

Maybe give a little more thought to the variable name `hasOnlyCliReporter`. 😅